File: pyg-manager-web/src/main/java/com/pyg/manager/controller/BrandController.java
package com.pyg.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pyg.manager.service.BrandService;
import com.pyg.pojo.TbBrand;

import com.pyg.utils.PageResult;
import com.pyg.utils.PygResult;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

/**
 * Created by on 2018/8/4.
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    //注入远程服务service对象
    @Reference(timeout = 1000000)
    private BrandService brandService;

    /**
     * 需求：查询所有品牌
     */
    @RequestMapping("findAll")
    public List<TbBrand> findAll() {

        //调用远程service服务方法
        List<TbBrand> brandList = brandService.findAll();

        return brandList;

    }

    /**
     * 需求：分页查询品牌数据
     */
    @RequestMapping("findPage/{page}/{rows}")
    public PageResult findPage(@PathVariable Integer page, @PathVariable Integer rows) {
        //调用远程服务service服务方法
        PageResult result = brandService.findPage(page, rows);
        return result;
    }

    /*
     * 需求：实现品牌数据添加
     */
    @RequestMapping("add")
    public PygResult add(@RequestBody TbBrand brand) {
        try {
            brandService.add(brand);

            //添加成功
            return new PygResult(true, "添加成功");

        } catch (Exception e) {
            e.printStackTrace();
            return new PygResult(false, "添加失败");
        }
    }

    /**
     * 需求：根据id查询品牌数据
     *
     * @param
     * @return
     */
    @RequestMapping("findOne/{id}")
    public TbBrand findOne(@PathVariable Long id) {
        TbBrand brand = brandService.findOne(id);
        return brand;
    }

    /*
     * 需求：实现品牌数据修改
     */
    @RequestMapping("update")
    public PygResult update(@RequestBody TbBrand brand) {
        try {
            brandService.update(brand);
            //添加成功
            return new PygResult(true, "添加成功");

        } catch (Exception e) {
            e.printStackTrace();
            return new PygResult(false, "添加失败");
        }
    }

    /*
     * 需求：实现品牌数据删除
     */
    @RequestMapping("del/{ids}")
    public PygResult delete(@PathVariable Long[] ids) {
        try {
            brandService.delete(ids);
            //删除成功
            return new PygResult(true, "删除成功");

        } catch (Exception e) {
            e.printStackTrace();
            return new PygResult(false, "删除失败");
        }
    }


    /**
     * 需求:查询品牌下拉列表
     * 查询数据格式：[{id:'1',text:'联想'},{id:'2',text:'华为'}]
     * 返回值：List<Map>
     *
     */
    @RequestMapping("findBrandList")
    public List<Map> findBrandList() {
        //调用远程service服务方法
        List<Map> brandList = brandService.findBrandList();
        return  brandList;
    }


}

File: pyg-shop-web/src/main/webapp/js/controller/brandController.js
//在模块下定义控制器
app.controller("brandController", function ($scope,$controller, brandService) {

    //控制器继承
    //继承baseController控制器，且把父控制器$scope传递子控制器的$scope
    $controller("baseController",{$scope:$scope});

    //定义函数
    //所有的函数都绑定$scope域中
    $scope.findAll = function () {
        //使用$http内置服务调用后台restfull请求
        brandService.findAll().success(function (data) {
            $scope.list = data;
        })
    };


    //定义分页查询方法
    $scope.findPage = function (page, rows) {
        //使用内置服务，发送分页查询请求
        brandService.findPage(page, rows).success(function (data) {
            $scope.list = data.rows;
            $scope.paginationConf.totalItems = data.total;
        })
    };

    //定义添加品牌数据方法
    $scope.add = function () {

        //接受添加，或者修改返回对象
        var objService = null;

        //否则就是添加
        objService = brandService.add($scope.entity);

        if ($scope.entity.id != null) {
            //修改
            objService = brandService.update($scope.entity);
        }


        //使用$http内置服务发送请求
        objService.success(function (data) {
            //判断
            if (data.success) {
                //重新刷新列表页面
                $scope.reloadList();
            } else {
                alert(data.message);
            }
        })

    };

    //定义根据id查询品牌数据方法
    $scope.findOne = function (id) {
        //内置服务发送请求
        brandService.findOne(id).success(function (data) {
            $scope.entity = data;
        })
    };


    //定义删除方法
    $scope.del = function () {
        //发送删除请求
        brandService.del($scope.selectIds).success(function (data) {
            //判断
            if (data.success) {
                //清空删除ids
                $scope.selectIds = [];
                //重新刷新列表页面
                $scope.reloadList();
            } else {
                alert(data.message);
            }
        })
    };


})
